lms/services/oauth2_token.py

```diff
@@ -51,19 +52,23 @@ def get(self):
 
         :raise OAuth2TokenError: if we don't have an OAuth 2 token for the user
         """
-        try:
-            return (
-                self._db.query(OAuth2Token)
-                .filter_by(
-                    application_instance=self._application_instance,
-                    user_id=self._user_id,
-                )
-                .one()
+        token = (
+            self._db.query(OAuth2Token)
+            .join(ApplicationInstance)
+            .filter(
+                # We don't query by application_instance but any token belonging to an AI with the same GUID
+                ApplicationInstance.tool_consumer_instance_guid
+                == self._application_instance.tool_consumer_instance_guid,
+                OAuth2Token.user_id == self._user_id,
             )
-        except NoResultFound as err:
-            raise OAuth2TokenError(
-                "We don't have an OAuth 2 token for this user"
-            ) from err
+            # We might find more that one token if we created them for multiple installs, prefere the latest.
```

Comment:
Ideally we'd solve this at the storage level, tokens (and files and groupings...) shouldn't be stored by applications instance but by GUID.

That involves DB schema changes, tricky data migrations which we should fix but for now we can get the correct behaviour with a query that shouldn't be that much expensive that the purely AI based one..